File: benchmarks/fixtures/webapp_py/utils/logging.py
"""Logging utilities."""

from config import LOG_LEVEL


class Logger:
    """Simple logger with level filtering."""

    def __init__(self, name: str, level: str = LOG_LEVEL):
        self.name = name
        self.level = level

    def info(self, message: str):
        """Log an info message."""
        self._write("INFO", message)

    def warning(self, message: str):
        """Log a warning message."""
        self._write("WARNING", message)

    def error(self, message: str):
        """Log an error message."""
        self._write("ERROR", message)

    def _write(self, level: str, message: str):
        """Write a log entry."""
        print(f"[{level}] {self.name}: {message}")


def get_logger(name: str) -> Logger:
    """Get a logger instance for the given name."""
    return Logger(name)

File: benchmarks/fixtures/webapp_php/utils/logging.php
<?php

namespace App\Utils;

/**
 * Minimal stdout logger.
 */
class Logger
{
    public function __construct(private string $name)
    {
    }

    public function info(string $message): void
    {
        // Emit an informational log line.
    }

    public function warn(string $message): void
    {
        // Emit a warning log line.
    }

    public function error(string $message): void
    {
        // Emit an error log line.
    }
}

/**
 * Logging facade.
 */
class Logging
{
    public static function getLogger(string $name): Logger
    {
        return new Logger($name);
    }
}
